File: project_workspace/RiskScanner.java
import ghidra.app.script.GhidraScript;
import ghidra.program.model.listing.Function;
import ghidra.program.model.listing.FunctionIterator;

public class RiskScanner extends GhidraScript {
    @Override
    public void run() throws Exception {
        println("\n--- [AI-ZeroDay] Starting Static Risk Scan ---");
        
        FunctionIterator iter = currentProgram.getFunctionManager().getFunctions(true);
        while (iter.hasNext() && !monitor.isCancelled()) {
            Function f = iter.next();
            if (f.getName().contains("strcpy")) {
                println("[ALERT] Dangerous Function Found: " + f.getName() + " at " + f.getEntryPoint());
            }
        }
        println("--- [AI-ZeroDay] Scan Complete ---\n");
    }
}

File: project_workspace/risk_scanner.py
from ghidra.app.script import GhidraScript

# List of dangerous functions to flag
BANNED = ["strcpy", "strcat", "gets", "sprintf"]

def run():
    print("\n--- [AI-ZeroDay] Starting Static Risk Scan ---")
    fm = currentProgram.getFunctionManager()
    funcs = fm.getFunctions(True) 
    
    for f in funcs:
        if f.getName() in BANNED:
            print("[ALERT] Dangerous Function Found: {} at {}".format(f.getName(), f.getEntryPoint()))
    print("--- [AI-ZeroDay] Scan Complete ---\n")

run()
